### PR TITLE
[Merged by Bors] - chore: rename `second_countable` -> `secondCountable` in lemma and instance names

### PR DESCRIPTION
**Pull request by grunweg**
While at it, also renames `sigma_compact`, `first_countable` and `locally_compact`, in the same way.

---

Best reviewed commit by commit.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by winstonyin** [comment]
Thanks for correcting these! Should some of the other things named `...second_countable...` be renamed for consistency, too? E.g. `ChartedSpace.secondCountable_of_sigma_compact`.

File: Mathlib/Topology/Compactness/SigmaCompact.lean [hunk 1]
```diff
@@ -175,15 +175,15 @@ lemma isSigmaCompact_iff_sigmaCompactSpace {s : Set X} :
   isSigmaCompact_iff_isSigmaCompact_univ.trans isSigmaCompact_univ_iff
 
 -- see Note [lower instance priority]
-instance (priority := 200) CompactSpace.sigma_compact [CompactSpace X] : SigmaCompactSpace X :=
+instance (priority := 200) CompactSpace.sigmaCompact [CompactSpace X] : SigmaCompactSpace X :=
```
Comment (j-loreaux) [inline Mathlib/Topology/Compactness/SigmaCompact.lean:178]: this one

File: Mathlib/Topology/Compactness/SigmaCompact.lean [hunk 2]
```diff
@@ -175,15 +175,15 @@ lemma isSigmaCompact_iff_sigmaCompactSpace {s : Set X} :
...
   ⟨⟨fun _ => univ, fun _ => isCompact_univ, iUnion_const _⟩⟩
 
 theorem SigmaCompactSpace.of_countable (S : Set (Set X)) (Hc : S.Countable)
     (Hcomp : ∀ s ∈ S, IsCompact s) (HU : ⋃₀ S = univ) : SigmaCompactSpace X :=
   ⟨(exists_seq_cover_iff_countable ⟨_, isCompact_empty⟩).2 ⟨S, Hc, Hcomp, HU⟩⟩
 
 -- see Note [lower instance priority]
-instance (priority := 100) sigmaCompactSpace_of_locally_compact_second_countable
+instance (priority := 100) sigmaCompactSpace_of_locallyCompact_secondCountable
```
Comment (j-loreaux) [inline Mathlib/Topology/Compactness/SigmaCompact.lean:186]: and this one

File: Mathlib/Topology/MetricSpace/ProperSpace.lean [hunk 3]
```diff
@@ -88,7 +88,7 @@ instance (priority := 100) proper_of_compact [CompactSpace α] : ProperSpace α
 
 -- see Note [lower instance priority]
 /-- A proper space is locally compact -/
-instance (priority := 100) locally_compact_of_proper [ProperSpace α] : LocallyCompactSpace α :=
+instance (priority := 100) locallyCompact_of_proper [ProperSpace α] : LocallyCompactSpace α :=
```
Comment (j-loreaux) [inline Mathlib/Topology/MetricSpace/ProperSpace.lean:91]: and this one

File: Mathlib/Topology/Metrizable/Urysohn.lean [hunk 4]
```diff
@@ -124,7 +124,7 @@ theorem exists_embedding_l_infty : ∃ f : X → ℕ →ᵇ ℝ, IsEmbedding f :
 /-- *Urysohn's metrization theorem* (Tychonoff's version): a T₃ topological space with second
 countable topology `X` is metrizable, i.e., there exists a metric space structure that generates the
 same topology. -/
-instance (priority := 90) metrizableSpace_of_t3_second_countable : MetrizableSpace X :=
+instance (priority := 90) metrizableSpace_of_t3_secondCountable : MetrizableSpace X :=
```
Comment (j-loreaux) [inline Mathlib/Topology/Metrizable/Urysohn.lean:127]: and this one

**Review by j-loreaux — COMMENTED** [review]
I think the explicitly named instances that I highlight below should have deprecations (that are *not* instances).

bors d+

**Comment by grunweg** [comment]
bors merge

**Pull request closed without merging**